crates/shared/src/order_validation.rs

```diff
@@ -1437,11 +1442,11 @@ mod tests {
             },
             ..creation
         };
-        let (order, quote) = validator
+        let (order, _) = validator
             .validate_and_construct_order(creation_, &domain_separator, Default::default(), None)
             .await
             .unwrap();
-        assert!(quote.is_some());
+        assert!(order.metadata.quote.is_some());
```

Comment:
We should probably keep both assertions.
```suggestion
        assert!(quote.is_some());
        assert!(order.metadata.quote.is_some());
```